### PR TITLE
Make staging secure endpoint toggle on staging

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
Staging versions of the apps are not able to connect with Plaid test credentials in the native apps because we always use the `.env.production` credentials there. This change is necessary so that we can ask QA (and internal testers) to test the VBA flow in TestFlight and Google Play store versions of the app.

### Fixed Issues
$ https://github.com/Expensify/App/issues/5075

### Tests

Tested that the app continued to work normally on dev builds + web. The specific change here is not easy to test locally since the app needs to be a "beta" version. But we can force it by applying this diff:

```diff
diff --git a/src/pages/settings/PreferencesPage.js b/src/pages/settings/PreferencesPage.js
index 2a659c3f2..93b36c317 100755
--- a/src/pages/settings/PreferencesPage.js
+++ b/src/pages/settings/PreferencesPage.js
@@ -101,26 +101,26 @@ const PreferencesPage = ({

                     {/* If we are in the staging environment then we have the option to switch from using the staging secure endpoint or the production secure endpoint. This enables QA */}
                     {/* and internal testers to take advantage of sandbox environments for 3rd party services like Plaid and Onfido */}
-                    {environment === CONST.ENVIRONMENT.STAGING && (
-                        <>
-                            <Text style={[styles.formLabel]} numberOfLines={1}>
-                                Test Preferences
-                            </Text>
-                            <View style={[styles.flexRow, styles.mb6, styles.justifyContentBetween]}>
-                                <View style={styles.flex4}>
-                                    <Text>
-                                        Use Secure Staging Server
-                                    </Text>
-                                </View>
-                                <View style={[styles.flex1, styles.alignItemsEnd]}>
-                                    <Switch
-                                        isOn={user.shouldUseSecureStaging || false}
-                                        onToggle={setShouldUseSecureStaging}
-                                    />
-                                </View>
+                    {/* {environment === CONST.ENVIRONMENT.STAGING && ( */}
+                    <>
+                        <Text style={[styles.formLabel]} numberOfLines={1}>
+                            Test Preferences
+                        </Text>
+                        <View style={[styles.flexRow, styles.mb6, styles.justifyContentBetween]}>
+                            <View style={styles.flex4}>
+                                <Text>
+                                    Use Secure Staging Server
+                                </Text>
+                            </View>
+                            <View style={[styles.flex1, styles.alignItemsEnd]}>
+                                <Switch
+                                    isOn={user.shouldUseSecureStaging || false}
+                                    onToggle={setShouldUseSecureStaging}
+                                />
                             </View>
-                        </>
-                    )}
+                        </View>
+                    </>
+                    {/* )} */}
                 </View>
             </View>
         </ScreenWrapper>
```
Keep in mind that if you do this then you will effectively be pointing the built app at the staging server and you won't be authenticated. So in order to test this you should also edit your `.env` to use the production credentials and then verify that the staging secure server is getting hit instead of production when following the rest of QA steps.

### QA Steps
1. With the app installed on the device navigate to settings
2. Tap "Preferences"
3. Verify that a new section called "Test Preferences" appears
4. Tap the "Use Secure Staging Server" toggle and verify it changes
5. Navigate back to Settings and then back to preferences and verify the change sticks
5. Navigate to [staging.new.expensify.com/bank-account](https://staging.new.expensify.com/bank-account/new)
1. Select the option to log in to bank account
1. Enter the test credentials (bank: `Chase` user: `user_good` pass: `pass_good`)
2. Verify there are no errors and you are able to continue to the next screen with these credentials

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [x] Android

### CODE REVIEW

**Comment by marcaaron** [comment]
Oh hmm actually just realizing that as a side effect of this change anyone using the beta version of the app effectively won't be able to set up a bank account properly. I'm not sure who we allow to access the beta or if this matters.

**Comment by marcaaron** [comment]
Making this a draft for now so that we can get some feedback on this https://github.com/Expensify/App/issues/5075#issuecomment-914525831

**Comment by marcaaron** [comment]
Alright this one's ready now.

**Review by ctkochan22 — APPROVED** [review]

File: src/libs/actions/User.js [hunk 1]
```diff
@@ -268,6 +268,13 @@ function subscribeToUserEvents() {
         });
 }
 
+/**
+ * @param {Boolean} shouldUseSecureStaging
+ */
+function setShouldUseSecureStaging(shouldUseSecureStaging) {
+    Onyx.merge(ONYXKEYS.USER, {shouldUseSecureStaging});
```
Comment (hoangzinh) [inline src/libs/actions/User.js:275]: Hi @marcaaron, is there a specific reason why you persisted `shouldUseSecureStaging` into `ONYXKEYS.USER`, instead of a new Onyx key? `shouldUseSecureStaging` doesn't seem related to the current user data.

Comment (marcaaron) [reply src/libs/actions/User.js:275]: Hi, Can you ask a more specific question? What do you mean by "related to the current user data"? This change is quite old, but I don't think there was any particular reason why it was put here vs. dedicated Onyx key.

Comment (marcaaron) [reply src/libs/actions/User.js:275]: It's a setting that the user can change so it was put on the `user` key. That's all I can say about it really.

Comment (hoangzinh) [reply src/libs/actions/User.js:275]: Thanks @marcaaron. The context is: It looks like we're working around to keep this data `shouldUseSecureStaging` in `ONYXKEYS.USER` when `Clear cache and restart` or Import Onyx or sign in/sign out. For example, this PR https://github.com/Expensify/App/pull/54673.

There is a proposal that suggests we should use a dedicated Onyx key, which I'm considering whether `shouldUseSecureStaging` makes sense to continue persisting at `ONYXKEYS.ACCOUNT` (this key replaces `ONYXKEYS.USER`), or we can have a dedicated Onyx key for it. https://github.com/Expensify/App/issues/69133#issuecomment-3223010428
